plugins/search/src/apis.ts

```diff
@@ -14,8 +14,17 @@
  * limitations under the License.
  */
 
+import { createApiRef, DiscoveryApi } from '@backstage/core';
 import { Entity, ENTITY_DEFAULT_NAMESPACE } from '@backstage/catalog-model';
+
 import { CatalogApi } from '@backstage/plugin-catalog-react';
+import { SearchQuery, SearchResultSet } from '@backstage/search-common';
+import qs from 'qs';
+
+export const searchApiRef = createApiRef<SearchApi>({
```

Comment:
Make `SearchApi` an interface type and call your implementation something like `SearchClient` instead. By using a class type here you make it tricky for other people of reimplement or decorate the API